jwst/associations/lib/rules_level2_base.py

```diff
@@ -794,6 +794,19 @@ def __init__(self):
                 ],
                 reduce=Constraint.notany
             ),
+            Constraint(
+                [
+                    DMSAttrConstraint(
+                        sources=['detector'],
+                        value='nrcblong'
+                    ),
+                    DMSAttrConstraint(
+                        sources=['exp_type'],
+                        value='nrc_tsgrism'
+                    )
+                ],
+                reduce=Constraint.notall
```

Comment:
Are you sure this should use `notall` as opposed to `notany`?  I have a hard time remembering exactly how the logic works.

Reply:
I think I'll have to defer to @stscieisenhamer on this, when he gets a chance - my thought was that exclusion is needed if both constraints are met, i.e. don't constrain nrc_tsgrism entries that have detector != nrcblong, but constrain entries with both - so not all constraints? But that goes against the logic in the constraint above the one I added that uses notany, so I may not have full comprehension...

Reply:
An explanation that satisfies me - the existing constraint using `notany` above the new constraint has two members, one of which looks for 'nirspec' in the detector column. I can't find an instance of a pool file with 'nirspec' listed in detector, just NRS1/NRS2 in nirspec's case. So as written, I think it's acting as a single constraint looking for 'filter'=='opaque' and excluding.

Reply:
Sounds like a bug to me. That 'nirspec' constraint should be looking at 'instrument', not 'detector'. But then the `notall` vs `notany` logic may need changing too.

Reply:
Should I fix it here or separate into another ticket?